docs/docs/developers/guides/js_apps/pay_fees.md

```diff
@@ -51,50 +56,96 @@ Create a new fee juice portal manager and bridge fee juice publicly to Aztec:
 
-### Gas Settings
+**The equivalent to specify fees via CLI...**
 
-#include_code gas_settings_vars yarn-project/stdlib/src/gas/gas_settings.ts javascript
+import { CLI_Fees } from '/components/snippets';
 
-import { Gas_Settings_Components, Gas_Settings } from '/components/snippets';
+<CLI_Fees />
 
-<Gas_Settings />
+### Claim and deploy
 
-<Gas_Settings_Components />
+Here we will use the `claim` object previously from the bridging section, and the corresponding `wallet`, to create the fee payment method. The payment method is then used to claim fee juice and pay for account contract deployment. Note the function used to bridge fee juice (private/public) should correspond to how the fee juice is claimed.
 
+```ts
+import { FeeJuicePaymentMethodWithClaim } from "@aztec/aztec.js";
+```
 
-### Other params
+#include_code claim_and_deploy yarn-project/bot/src/factory.ts javascript
 
-Fee and gas padding params can be left to their default values, and the estimateGas boolean can be used when simulating a tx to calculate gas.
+**The equivalent to specify fees via CLI...**
 
-With the fee options explained, lets do a paid transaction.
+The CLI tool `aztec-wallet` takes the fee payment method via the param: `--payment method=fee_juice,claim`
 
-## Examples
+#### Claim and Pay
+
+Claiming bridged fee juice and using it to pay for transaction fees results in fees being paid publicly.
+
+Calling a function, in this case checking the balance of the fee juice contract:
+
+#include_code claim_and_pay yarn-project/end-to-end/src/e2e_fees/fee_juice_payments.test.ts javascript
+
+### Fee Paying Contract
+
+Users can pay fees privately or publicly via FPCs. A FPC must be deployed and funded with fee juice before it can be used to pay for transactions. Users must have the token that the FPC accepts to pay for transactions. They must have a private balance of the token if they want to pay for fees privately, and a public balance if they want to pay for fees publicly.
+
+Import the appropriate method from aztec.js.
+
+```ts
+import {
+  PrivateFeePaymentMethod,
+  PublicFeePaymentMethod,
+} from "@aztec/aztec.js";
+```
+
+The fee payment method is created and used as follows, with similar syntax for private or public fee payments:
+
+#include_code fpc yarn-project/end-to-end/src/e2e_fees/public_payments.test.ts javascript
+
+In this example, thanks to the FPC's `accepted_asset` being banana tokens, Alice only needs to hold this token and not fee juice. The asset that a FPC accepts for paying fees is determined when the FPC is deployed. The function being called happens to also be a transfer of banana tokens to Bob.
+
+More on FPCs [here](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/noir-contracts/contracts/fpc_contract/src/main.nr)
+
+See this [section](../../reference/environment_reference/cli_wallet_reference.md#fee-paying-contract) for paying fees via an FPC using the CLI wallet.
```

Comment:
This link is going to break until this pr is merged. https://github.com/AztecProtocol/aztec-packages/pull/12884/files#